CMakeLists.txt

```diff
@@ -152,6 +152,103 @@ IF (INCLUDE_SPI)
 	ENDIF (HAVE_LINUX_SPI_H)
 ENDIF (INCLUDE_SPI)
 
+
+# Depending on available packages, JavaScript files stored in plain format can
+# be minified (or uglyfied...), ie: removing all useless caracters as spaces, CRs, tabs...
+# and variables names shortened to one caracter in order to minimize memeory footprint and increase 
+# global performance. We use uglifyjs (which does not compress in anyway despite its name...)
+# Do it only if not compiling in debug mode
+
+
+# Buil list of JS files in www dir and subdirs to be compressed and minified on install
+file(GLOB_RECURSE js_in_files RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}/www/" *.js)
+
+option(MINIFY_JS "Minify JavaScript files" YES)
+unset(UGLY CACHE)
+unset(GZIP CACHE)
+unset (PARAM CACHE)
+#find_program(UGLY uglifyjs)
+find_program(GZIP gzip)
+
+IF (${CMAKE_BUILD_TYPE} STREQUAL "Debug")
```

Comment:
I don't think we can mangle variables at all. Specially when we are using more AngularJS code.
Variables from other JS files are used in another JS file... this can go terribly wrong.
I think we should remove the whole renaming of variables
It is also not needed, it's an opensource project, and we do not need to protect the code